sunpy/util/xml.py

```diff
@@ -124,3 +123,63 @@ def get_node_text(node):
         else:
             raise NotTextNodeError
     return t
+
+def xml_comments_to_dict(xmlstring):
```

Comment:
Do we need two functions?

Reply:
Actually to traverse through any XML tree (since it's a utility function and not just for jp2 files), we need to use a recursive function. For that we need a recursive function. I have followed the same convention that the other function in `xml.py` which is `xml_to_dict` uses.